### ISSUE TITLE
Question about the implementation of VGG19 Perceptual Loss

### ISSUE THREAD

**Issue by shuuki4** [NVIDIA/pix2pixHD#136]
According to the implementation of perceptual loss (`VGGLoss`), this VGG19 network directly uses the model structure & pretrained weights from `torchvision.models.VGG19`.
However, torchvision documents (https://pytorch.org/docs/stable/torchvision/models.html) state that the input to the VGG19 network should be normalized using `mean = [0.485, 0.456, 0.406]` and `std = [0.229, 0.224, 0.225].` But current implementation puts an image that is simply normalized in the scale of `-1 ~ 1`.
Isn't it necessary to renormalize the image input before feeding them into VGG19 network?

**Comment by mrgloom** [comment]
Do we need resize image to 224x224(like in paper https://arxiv.org/pdf/1409.1556.pdf) before feeding it to VGG19?

cc: @tcwang0509

**Comment by Jarvisss** [comment]
> According to the implementation of perceptual loss (`VGGLoss`), this VGG19 network directly uses the model structure & pretrained weights from `torchvision.models.VGG19`.
> However, torchvision documents (https://pytorch.org/docs/stable/torchvision/models.html) state that the input to the VGG19 network should be normalized using `mean = [0.485, 0.456, 0.406]` and `std = [0.229, 0.224, 0.225].` But current implementation puts an image that is simply normalized in the scale of `-1 ~ 1`.
> Isn't it necessary to renormalize the image input before feeding them into VGG19 network?

Did you reimplement the perceptual loss and normalize the way like torchvision documents?

**Comment by rafaelbou** [comment]
Moreover, I found that the VGG loss isn't decreasing during training.
I was expected that it will improve as the output improve.

Does anyone encountered this issue?

**Comment by mohammadhosseinashoori** [comment]
> 
> 
> According to the implementation of perceptual loss (`VGGLoss`), this VGG19 network directly uses the model structure & pretrained weights from `torchvision.models.VGG19`.
> However, torchvision documents (https://pytorch.org/docs/stable/torchvision/models.html) state that the input to the VGG19 network should be normalized using `mean = [0.485, 0.456, 0.406]` and `std = [0.229, 0.224, 0.225].` But current implementation puts an image that is simply normalized in the scale of `-1 ~ 1`.
> Isn't it necessary to renormalize the image input before feeding them into VGG19 network?

hello, Did you find the answer to this question?
**Isn't it necessary to renormalize the image input before feeding them into VGG19 network?**
This is still a question for me!
